crates/key-value-azure/src/store.rs

```diff
@@ -158,24 +159,35 @@ impl Store for AzureCosmosStore {
         self.client
             .create_document(pair)
             .is_upsert(true)
+            .allow_tentative_writes(TentativeWritesAllowance::Allow)
             .await
             .map_err(log_error)?;
         Ok(())
     }
 
     async fn delete(&self, key: &str) -> Result<(), Error> {
-        if self.exists(key).await? {
-            let document_client = self
-                .client
-                .document_client(key, &self.store_id)
-                .map_err(log_error)?;
-            document_client.delete_document().await.map_err(log_error)?;
+        let document_client = self
+            .client
+            .document_client(key, &self.store_id.clone().unwrap_or(key.to_string()))
```

Comment:
Sanity check... `store_id` seems to be a partition key (or PK column???) but `key` seems to be a row key - is it correct to mix them?  The docs imply the default store id is `/id` rather than the current row key, but I may be misunderstanding how the store ID gets applied.

_EDIT: oh is this aligning with `Pair`'s implementation of `CosmosEntity::partition_key`?  that feels like we should make the relationship clearer then.  Maybe improve docs on `self.store_id` too?_

Reply:
I updated the `store_id` inline doc to explain that `/id` is the row key:
```
    /// An optional store id to use as a partition key for all operations.
    ///
    /// If the store id not set, the store will use `/id` (the row key) as the
    /// partition key. For example, if `store.set("my_key", "my_value")` is
    /// called, the partition key will be `my_key` if the store id is set to
    /// `None`. If the store id is set to `Some("myappid/default"), the
    /// partition key will be `myappid/default`.
    store_id: Option<String>,
}
```